### PR TITLE
koordlet: fix node memory collection

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

- Fix a bug of the metrics advisor in which the node memory usage in NodeMetric shows as kilo-bytes while the other fields use bytes.
- Add UTs for node resource collector to avoid regression.
- Revise logs and error handling in the states informer.

### Ⅱ. Does this pull request fix one issue?

Fixed the bug in koordlet's node resource collector (introduced in #1316).

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/statesinformer/states_nodemetric.go [hunk 1]
```diff
@@ -483,7 +482,7 @@ func (r *nodeMetricInformer) collectNodeGPUMetric(queryparam metriccache.QueryPa
 }
 
 func (r *nodeMetricInformer) collectNodeAggregateMetric(endTime time.Time, aggregatePolicy *slov1alpha1.AggregatePolicy) []slov1alpha1.AggregatedUsage {
-	aggregateUsages := []slov1alpha1.AggregatedUsage{}
+	var aggregateUsages []slov1alpha1.AggregatedUsage
```
Comment (zwzhang0107) [inline pkg/koordlet/statesinformer/states_nodemetric.go:485]: make([], )

Comment (saintube) [reply pkg/koordlet/statesinformer/states_nodemetric.go:485]: It's OK in Golang to append a nil slice.

**Review by jasonliu747 — APPROVED** [review]

**Review by LambdaHJ — COMMENTED** [review]
/lgtm

**Review by hormes — APPROVED** [review]

**Pull request merged**